### PR TITLE
New Build.Bat to Detect VS Install and Use it's MSBuild - HydraBuild (Ready to Merge)

### PR DESCRIPTION
**Pull request by CDAGaming**
Features:

- Adds New Build.bat to detect and build Automatically based on VS Version

### CODE REVIEW

File: build.bat [hunk 1]
```diff
@@ -1,4 +1,23 @@
 echo off
```
Comment (Furtif) [inline build.bat]: + nuget.exe restore "NecroBot-Private for Pokemon GO.sln"

**Comment by Furtif** [comment]
@echo off
nuget.exe restore "NecroBot-Private for Pokemon GO.sln"
set necrobuilder=""
for /f "delims=" %%i in ('dir /s /b /a-d "%programfiles(x86)%\MSBuild.exe"') do (
if %necrobuilder%=="" (set necrobuilder="%%i")
)
%necrobuilder% "NecroBot-Private for Pokemon GO.sln" /property:Configuration="Release" /property:Platform="Any CPU"
echo %necrobuilder%
set necrobuilder=

File: build.bat [hunk 2]
```diff
@@ -1,4 +1,23 @@
...
+
+:VS2015_Community
+echo Building under Visual Studio 2015 Community as %processor_architecture%
+if "%processor_architecture%"=="AMD64" (
+nuget.exe restore "NecroBot-Private for Pokemon GO.sln"
+"C:\Program Files (x86)\MSBuild\14.0\Bin\MSBuild.exe" "NecroBot-Private for Pokemon GO.sln" /property:Configuration=Release /property:Platform=x64
+) ELSE (
+nuget.exe restore "NecroBot-Private for Pokemon GO.sln"
+"C:\Program Files (x86)\MSBuild\14.0\Bin\MSBuild.exe" "NecroBot-Private for Pokemon GO.sln" /property:Configuration=Release /property:Platform=x86
+)
```
Comment (Furtif) [inline build.bat]:
```
:end
```

File: build.bat [hunk 3]
```diff
@@ -1,4 +1,23 @@
...
+echo Building under Visual Studio 2017 Community as %processor_architecture%
+if "%processor_architecture%"=="AMD64" (
 nuget.exe restore "NecroBot-Private for Pokemon GO.sln"
-"C:\Program Files (x86)\MSBuild\14.0\Bin\MsBuild.exe" "NecroBot-Private for Pokemon GO.sln" /property:Configuration=Release /property:Platform=x86
+"C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\MSBuild.exe" "NecroBot-Private for Pokemon GO.sln" /property:Configuration=Release /property:Platform=x64
+) ELSE (
+nuget.exe restore "NecroBot-Private for Pokemon GO.sln"
+"C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\MSBuild.exe" "NecroBot-Private for Pokemon GO.sln" /property:Configuration=Release /property:Platform=x86
+)
+
```
Comment (Furtif) [inline build.bat]:
```
goto end
```

**Comment by CDAGaming** [comment]
Will do Review Suggestions very shortly

**Comment by hardknoxni69a** [comment]
should make anycpu by default for bat so people can always get whats best for them

**Comment by hardknoxni69a** [comment]
please fix conflicts

**Comment by Lord-Haji** [comment]
@hardknoxni69a conflicts ? I don't see any conflicts

**Comment by CDAGaming** [comment]
Me neither, but Appveyor build error is because of Appveyor being a douche, will fix shortly

**Pull request merged**